source/GameGraphRoot.cpp

```diff
@@ -546,6 +582,71 @@ void GameGraphRoot::update(float timestep) {
 	}
 }
 
+/**
+ * Returns true iff a button was properly tapped (the tap event both started and ended on the
+ * button)
+ *
+ * @param button The button
+ * @param tapData The start and end locations provided by the input controller
+ */
+bool tappedBtn(std::shared_ptr<Button> button, std::tuple<Vec2, Vec2> tapData) {
+	return button->containsScreen(std::get<0>(tapData)) &&
+		   button->containsScreen(std::get<1>(tapData));
+}
+
```

Comment:
This method has been factored out into `ButtonManager`. Please call it from there.